src/audio/dai.c

```diff
@@ -769,10 +769,31 @@ static int dai_config(struct comp_dev *dev, struct sof_ipc_dai_config *config)
 			dd->chan = NULL;
 		}
 		break;
+	case SOF_DAI_INTEL_ALH:
+		/* set to some non-zero value to satisfy the condition below,
+		 * it is recalculated in dai_params() later
+		 */
+		dd->frame_bytes = 4;
+
+		/* SDW HW FIFO requires 32bit MSB aligned sample data for
+		 * all formats, such as 8/16/24/32 bits. Now set frame_fmt
+		 * to S32_LE to let volume component do the conversion.
```

Comment:
the comment is a bit misleading, the dai should not assume that there's a volume component but only that it deals with data that has been left-aligned, and throw an error if this is not the case,